### PR TITLE
8253756: C2 CompilerThread0 crash in Node::add_req(Node*)

### PR DESCRIPTION
**Pull request by rwestrel**
The outer strip mined loop is initially created as a skeleton and then
expanded once loop opts are over. As long as it is a skeleton, the
OuterStripMinedLoopEndNode cannot be constant folded because its input
is a place holder. So OuterStripMinedLoopEndNode::Value() blocks
constant folding. This bug triggers because the
OuterStripMinedLoopEndNode input after expansion is a constant but the
OuterStripMinedLoopEndNode is not constant folded (i.e. it's a dead
loop but it stays in the graph). The fix I propose is to change the
behavior OuterStripMinedLoopEndNode::Value() so it blocks constant
folding only until expansion but not after.

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Testing

|     | Linux x64 | Windows x64 | macOS x64 |
| --- | ----- | ----- | ----- |
| Build | ✔️ (3/3 passed) | ✔️ (2/2 passed) | ✔️ (2/2 passed) |
| Test (tier1) | ✔️ (9/9 passed) | ❌ (1/9 failed) | ✔️ (9/9 passed) |

**Failed test task**
- [Windows x64 (langtools/tier1)](https://github.com/rwestrel/jdk/runs/1221555809)

### Issue
 * [JDK-8253756](https://bugs.openjdk.java.net/browse/JDK-8253756): C2 CompilerThread0 crash in Node::add_req(Node*)

### Reviewers
 * [Vladimir Ivanov](https://openjdk.java.net/census#vlivanov) (@iwanowww - **Reviewer**)
 * [Tobias Hartmann](https://openjdk.java.net/census#thartmann) (@TobiHartmann - **Reviewer**)

### Download
`$ git fetch https://git.openjdk.java.net/jdk pull/537/head:pull/537`
`$ git checkout pull/537`

### CODE REVIEW

File: src/hotspot/share/opto/loopnode.cpp [hunk 1]
```diff
@@ -1726,16 +1726,38 @@ void OuterStripMinedLoopNode::adjust_strip_mined_loop(PhaseIterGVN* igvn) {
     igvn->replace_node(outer_le, iff);
     inner_cl->clear_strip_mined();
   }
+  igvn->C->print_method(PHASE_DEBUG, 2);
```
Comment (iwanowww) [inline src/hotspot/share/opto/loopnode.cpp:1729]: Looks like a leftover from debugging. Otherwise, the fix looks good.

**Review by iwanowww — CHANGES_REQUESTED** [review]

File: src/hotspot/share/opto/loopnode.cpp (hunk 1 above)
Comment (rwestrel) [reply src/hotspot/share/opto/loopnode.cpp]: Good catch. I thought I removed it but obviously didn't. I updated the change. Thanks for the review.

**Review by iwanowww — APPROVED** [review]
Looks good.

**Review by TobiHartmann — APPROVED** [review]
Looks reasonable to me.

**Pull request closed without merging**